source/dofs/dof_handler_policy.cc

```diff
@@ -4359,18 +4362,80 @@ namespace internal
       template <class DoFHandlerType>
       NumberCache
       ParallelDistributed<DoFHandlerType>::
-      renumber_mg_dofs (const unsigned int                          /*level*/,
-                        const std::vector<types::global_dof_index> &/*new_numbers*/) const
+      renumber_mg_dofs (const unsigned int                          level,
+                        const std::vector<types::global_dof_index> &new_numbers) const
       {
-        // this is not currently implemented, but should be simple to do by
-        // just calling the function like in the sequential case just with
-        // an appropriate index set argument
-        Assert(false, ExcNotImplemented());
+        // we only implement the case where the multigrid numbers are
+        // renumbered within the processor's partition, rather than the most
+        // general case
+        const std::vector<IndexSet> index_sets = dof_handler->locally_owned_mg_dofs_per_processor(level);
+
+        constexpr int dim = DoFHandlerType::dimension;
+        constexpr int spacedim = DoFHandlerType::space_dimension;
+        const parallel::Triangulation<dim, spacedim> *tr =
+          (dynamic_cast<const parallel::Triangulation<dim, spacedim>*>
+           (&this->dof_handler->get_triangulation()));
+        Assert(tr != nullptr, ExcInternalError());
+        const unsigned int my_rank = Utilities::MPI::this_mpi_process(MPI_COMM_WORLD);
```

Comment:
i think you should ask for a communicator, just in case it's not MPI_COMM_WORLD which is used in p:;d::Tria

Reply:
Of course ;-)

Reply:
essentially all three comments of mine are within the "of course" / typo realm 😄